src/harness/testcases/WINNF_FT_S_REG_testcase.py

```diff
@@ -0,0 +1,266 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import unittest
+
+import sas
+from util import winnforum_testcase
+
+
+class RegistrationTestcase(unittest.TestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_REG_1(self):
+    """Array Multi-Step registration for CBSDs (Cat A and B).
+
+    The response should be SUCCESS.
+    """
+
+    # Load Devices
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+
+    # Pre-load conditionals
+    conditionals_a = {
+        'cbsdCategory': device_a['cbsdCategory'],
+        'fccId': device_a['fccId'],
+        'userId': device_a['userId'],
+        'cbsdSerialNumber': device_a['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam']
+    }
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'userId': device_b['userId'],
```

Comment:
userID is a "Required" parameter and not "Reg-Conditional" do we need to include it as part of the conditional parameters that get loaded into the SAS (as in the CPI would be loading them)?

Reply:
Yes, The SAS UUT needs the user ID, FCC ID, and serial # so that it can link the conditional reg. info with the device itself. I agree the code could be a little clearer, but we intentionally made only minimal changes to the existing code.

Reply:
It needs the FCC ID and serial number (that makes up the CBSD ID) but it should not need the user ID.  The test is trying to emulate the scenario where a CPI inserts data into the SAS and then a registration occurs.  When a CPI enters data into the SAS they will enter REG-Conditional data and the user ID is not part of that.  It's quite possible the the CPI entering the data doesn't even know what the user ID that will register the CBSD later would be

Reply:
User ID is a required field in RegistrationRequest according to the SAS-CBSD TS, so it needs to be present here. (The only other required fields are serial number and FCC ID.)

Reply:
the User ID will be included in the RegistrationRequest (as is required) but I can't find anywhere where it says that it must be provided by the CPI when the other Reg-Conditional parameters are provided.  

The TC (using Reg1 as an example) says 

"All REGConditional
parameters for the three CBSDs (a combination of both Cat A and
Cat B) should already be pre-loaded into the SAS."

User Id is required (but not Reg-Conditional) therefore it should be provided in the registrationirequest message but it is not needed as part of the "load conditional parameters into the SAS"

When you are marrying the incomplete registration-request with the data preloaded into the SAS you can do so using just FCC-ID and serial number.

It's possible that there is a CPI document somewhere that says that user ID must be loaded with the other Reg-Conditionals and I just haven't seen it yet; if so can someone point me to it?

Reply:
Oops, sorry, I was too eager to reply to your comment and missed which line this was referencing :)

It's also not clear to me whether user ID is currently needed, but it feels likely that it would be. I suspect WG5 is still working out those details. I propose we pick one path for now and add an item to the issue tracker (https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System/issues) to look into this and fix any code, if needed. Would you be OK to leave the proposed code as-is for now, pending resolution of this issue?

Reply:
sounds great to me!

The only reason I feel the user Id doesn't need to be in there is because we are mimicking how things would work when a CPI inputs data into the SAS and i don't think CPIs are going to have access to user ID but they will know the FCC ID and the serial number of the CBSD they are entering data for.

Reply:
Thanks! Issue created: https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System/issues/290